ddl/cluster.go

```diff
@@ -111,12 +123,21 @@ func ValidateFlashbackTS(ctx context.Context, sctx sessionctx.Context, flashBack
 		}
 		currentTS = currentVer.Ver
 	}
-	if oracle.GetTimeFromTS(flashBackTS).After(oracle.GetTimeFromTS(currentTS)) {
+	oracleFlashbackTS := oracle.GetTimeFromTS(flashBackTS)
+	if oracleFlashbackTS.After(oracle.GetTimeFromTS(currentTS)) {
 		return errors.Errorf("cannot set flashback timestamp to future time")
 	}
-	if oracle.GetTimeFromTS(flashBackTS).After(expression.GetMinSafeTime(sctx)) {
-		return errors.Errorf("cannot set flashback timestamp to too close to present time")
+
+	start := time.Now()
+	minSafeTime := getStoreGlobalMinSafeTS(sctx.GetStore())
+	for oracleFlashbackTS.After(minSafeTime) {
+		if time.Since(start) >= FlashbackGetMinSafeTimeTimeout {
+			return errors.Errorf("cannot set flashback timestamp after min-resolved-ts(%s)", minSafeTime)
+		}
+		time.Sleep(time.Second)
+		minSafeTime = getStoreGlobalMinSafeTS(sctx.GetStore())
```

Comment:
What about using `select` here to watch a time ticker rather than `time.Sleep`? Then we can also watch the `<-ctx.Done()` to prevent from goroutine leak.

Reply:
Done, PTAL